source/Camera/Advanced/MultiCameraCaptureInParallel/MultiCameraCaptureInParallel.cpp

```diff
@@ -0,0 +1,74 @@
+/*
+This example shows how to capture point clouds with multiple cameras in parallel.
+*/
+
+#include <Zivid/Zivid.h>
+
+#include <chrono>
+#include <future>
+#include <iostream>
+#include <mutex>
+#include <vector>
+
+namespace
+{
+    void captureInThread(Zivid::Camera &camera, std::mutex &m)
+    {
+        const auto parameters = Zivid::CaptureAssistant::SuggestSettingsParameters{
+            Zivid::CaptureAssistant::SuggestSettingsParameters::AmbientLightFrequency::none,
+            Zivid::CaptureAssistant::SuggestSettingsParameters::MaxCaptureTime{ std::chrono::milliseconds{ 800 } }
+        };
+        const auto settings = Zivid::CaptureAssistant::suggestSettings(camera, parameters);
+
+        // Capturing frame
+        const auto frame = camera.capture(settings);
+
+        const auto dataFile = "Frame_" + camera.info().serialNumber().value() + ".zdf";
+
+        // Saving frame to file
+        std::unique_lock<std::mutex> lock(m);
+        frame.save("Frame_" + camera.info().serialNumber().value() + ".zdf");
```

Comment:
Is there a reason that you add a mutex around frame.save()? I don't think that should be needed.
You have made the variable `dataFile` above, but you are not using it here.

Reply:
It crashes without a mutex. netCDF doc says not all netCDF calls are thread safe.

Reply:
I don't know exactly what crashes. I only see a netCDF function on the top of the call stack when this segfault happens

Reply:
on my laptop it crashes without the mutex. Not sure the reason

Reply:
@robertkt1 OK; can you make an SDK issue for this?

Reply:
@apartridge I could, but I am not sure what the issue would be. If Karthik says not all netCDF calls are thread safe then I don't see this as a bug? 
Or is it that we want all our (one-camera-)api to be thread safe?
I haven't tested it, but if the one-camera-api should be thread safe I assume we want to test downsampling and transformation in multi threads as well.

Reply:
Our official position is that one can use multiple cameras at the same time from different threads, as long as it is one thread per camera (and that you connect to cameras sequentially). We don't say anything in particular about saving frame from multiple cameras at the same time, but from an user perspective I would expect that to work (same as other frame/point cloud operations like copy data to CPU, downsampling, transform etc). So I think we should make a bug for this, so that we can improve multi-threading behavior in the future. If netCDF itself is not thread safe, that just means we must add locking on SDK side. NetCDF should not dictate our multi-threading policy.